index.js

```diff
@@ -505,14 +511,72 @@ class XiaomiRoborockVacuum {
         await this.device.activateCleaning();
       } else if (!state) { // Stop cleaning
         this.log.info(`ACT setCleaning | ${this.model} | Stop cleaning and go to charge.`);
-        await this.device.activateCharging(); // Charging works for 1st, not for 2nd
+        await this.activateCharging(); // Charging works for 1st, not for 2nd
```

Comment:
Why not using the `device` implementation? It stops and the calls charge. Wouldn’t it be safer? If not needed to stop, maybe we can fix it at the `miio` library straight away?
I like in that library it actually checks for the `ok` answer to the command (which we are not doing in here).

Reply:
The S5 and S6 won't go back to dock with the miio Call. (first stop then charge) .. so i tested with directly calling app_charge and this work.

I think there is something different after S1

Reply:
Yes... that's a key point! We cannot break the behaviour for old/other models that might be using this... But I'm happy to merge this and explore our options if this fixes it for 80% of the people.

When I mentioned the `miio` library, I meant in [here](https://github.com/nicoh88/miio/blob/master/lib/devices/vacuum.js#L202-L210) (should you want to try updating that library, although some times I feel like completely moving the library into this project, since that npm module is not actively developed anymore).